mockkubeapiserver/kubernetes_builtin_schema.go

```diff
@@ -5,35 +5,79 @@ import (
 	"sync"
 
 	"sigs.k8s.io/structured-merge-diff/v4/typed"
+	"sigs.k8s.io/yaml"
 
 	_ "embed"
 )
 
 //go:embed kubernetes_builtin_schema.yaml
 var kubernetesBuiltinSchemaYAML string
 
+//go:embed kubernetes_builtin_schema.meta.yaml
+var kubernetesBuiltinSchemaMetaYAML string
+
+const BuiltinKey = "kubernetes_builtin_schema"
+
 type Schema struct {
 	Parser *typed.Parser
+	Meta   *SchemaMeta
 }
 
-var cacheKubernetesBuiltinSchema *Schema
+type SchemaMeta struct {
+	Resources []SchemaMetaResource `json:"resources"`
+}
 
-var mutexKubernetesBuiltinSchema sync.Mutex
+type SchemaMetaResource struct {
+	Key      string `json:"key"`
+	Group    string `json:"group"`
+	Version  string `json:"version"`
+	Kind     string `json:"kind"`
+	Resource string `json:"resource"`
+	Scope    string `json:"scope"`
+}
+
+type schemaCache struct {
+	mutex   sync.Mutex
+	schemas map[string]*Schema
+}
+
+var globalSchemaCache schemaCache
 
 func KubernetesBuiltInSchema() (*Schema, error) {
-	mutexKubernetesBuiltinSchema.Lock()
-	defer mutexKubernetesBuiltinSchema.Unlock()
+	return globalSchemaCache.Get(BuiltinKey)
+}
+
+func (c *schemaCache) Get(key string) (*Schema, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
 
-	if cacheKubernetesBuiltinSchema != nil {
-		return cacheKubernetesBuiltinSchema, nil
+	if schema := c.schemas[key]; schema != nil {
+		return schema, nil
 	}
 
-	schemaYAML := kubernetesBuiltinSchemaYAML
+	if key != BuiltinKey {
```

Comment:
I don't get it, if the key is NOT "BuiltInKey", then this will fail, but then how do you get anything but "BuiltinKey" in the list?

Reply:
That's the only one that works at the moment.  But I thought this was clearer (externally) and can allow for more in future...  It's not great, but maybe it's better than the multiple globals we had before.